### ISSUE TITLE
Development proxy ignores path in K8s API URL

### ISSUE THREAD

**Issue by rbq** [operator-framework/operator-sdk#4925]
#### What did you do?

1. set `~/.kube/config` to a Rancher-managed cluster's API  
    (sth. like `https://my-domain.tld/k8s/clusters/c-some-id`)
2. `kubectl apply […]` to deploy CRD
3. `operator-sdk run` to run operator

#### What did you expect to see?

I expected both steps 2 and 3 to target the cluster configured in step 1.

#### What did you see instead? Under which circumstances?

1. `kubectl apply` works as expected
2. `operator-sdk run` ignores the path component when setting up the proxy and uses `https://my-domain.tld` instead

In my particular case this didn't actually surface as an error. Instead the modified API endpoint without a path happened to target the management cluster of my Rancher deployment instead of a development cluster. So it actually ran the operator against a different cluster without any warning at all.

#### Environment

**Operator type:**

/language ansible

**Kubernetes cluster type:**

Rancher, RKE

`$ operator-sdk version`

operator-sdk version: "v1.7.1-19-g3ce1eeef", […], kubernetes version: "v1.19.4", […]

`$ kubectl version`

Client Version: version.Info{Major:"1", Minor:"21", […]}
Server Version: version.Info{Major:"1", Minor:"20", […]}

#### Possible Solution

1. To prevent people from shooting themselves in the foot I'd suggest that `operator-sdk run` (and any other commands that might behave similarly) fail with an error message if they encounter an API endpoint with a path component.
2. It would be nice if Operator SDK could be improved to work with clusters that have an API with a path component, such as the ones behind a Rancher instance or other proxies/multiplexers.

#### Additional context

* https://github.com/kubernetes/kubernetes/issues/94280
* https://kubernetes.slack.com/archives/CAW0GV7A5/p1620896783095000

**Comment by laxmikantbpandhare** [comment]
Fix for this issue is dependent on [kubectl PR](https://github.com/kubernetes/kubernetes/pull/97350). Once this PR is merged then only we will be able to move ahead with this fix as it has changes in **apimachinery**.

**Comment by laxmikantbpandhare** [comment]
Created another issue #4980 which will fix this issue and throw an error for now. Once [kubectl PR](https://github.com/kubernetes/kubernetes/pull/97350) gets merged then we will move ahead with changes.

**Comment by rashmigottipati** [comment]
Removing this from v1.9.0 milestone as #4980 (in milestone v1.9.0) is tracking this issue.

**Comment by laxmikantbpandhare** [comment]
As part of the Operator SDK v1.9.0 release, we have moved temporary fix for this issue. It will throw an error if API endpoint has Path in it. @rbq Could you please try with new version of Operator SDK. You can download the release and view the changes here: https://github.com/operator-framework/operator-sdk/releases/tag/v1.9.0

**Comment by rbq** [comment]
> As part of the Operator SDK v1.9.0 release, we have moved temporary fix for this issue. It will throw an error if API endpoint has Path in it. @rbq Could you please try with new version of Operator SDK. You can download the release and view the changes here: https://github.com/operator-framework/operator-sdk/releases/tag/v1.9.0

Checked, throws an error as expected. :)

**Comment by laxmikantbpandhare** [comment]
@rbq - thank you for you confirmation :)

**Comment by laxmikantbpandhare** [comment]
Till now, we did the temporary fix for this issue as this issue was dependent on [kubectl PR](https://github.com/kubernetes/kubernetes/pull/97350). [kubectl PR](https://github.com/kubernetes/kubernetes/pull/97350) got merged and we are moving ahead with the permanent fix for this issue.

**Comment by laxmikantbpandhare** [comment]
@rbq - I wanted to re-create this issue on my local machine. Wanted to know about the steps you did. I am working RKE cluster for the first time.

**Comment by laxmikantbpandhare** [comment]
> @laxmikantbpandhare I don't know if you can actually use your local machine to run both Rancher and a cluster node, but the steps should be:
> 
> 1. [run Rancher as a single container](https://rancher.com/docs/rancher/v2.5/en/installation/other-installation-methods/single-node-docker/)
> 2. [provision a cluster via Rancher](https://rancher.com/docs/rancher/v2.5/en/cluster-provisioning/rke-clusters/)
> 3. [download the kubeconfig file](https://rancher.com/docs/rancher/v2.5/en/cluster-admin/cluster-access/kubectl/#accessing-clusters-with-kubectl-from-your-workstation)
> 
> The kubeconfig file will instruct clients to access the provisioned cluster through Rancher using a URL that contains a path.

Great, I will try this. thank you.

**Comment by jmrodri** [comment]
This was waiting on k8s 1.23 release per the [above comment](https://github.com/operator-framework/operator-sdk/issues/4925#issuecomment-942499431). We merged k8s 1.23 in PR #5505

**Comment by rbq** [comment]
I'm getting the message "[...] contains a path component, which the proxy server is currently unable to handle properly. Work on this issue is being tracked here: [...]" in 1.19.0.

**Comment by laxmikantbpandhare** [comment]
thank you for bringing this @rbq - We are working on it and will fix this soon.

**Comment by laxmikantbpandhare** [comment]
@rbq - We removed the change that was throwing an error when there is a path in the proxy server. It will be available in the 1.20 `operator-sdk` release.

**Comment by rbq** [comment]
@laxmikantbpandhare That's great to hear, thanks a lot!

**Comment by rbq** [comment]
@laxmikantbpandhare Are releases cut based on schedule (the milestone says due today) or features?

**Comment by laxmikantbpandhare** [comment]
@rbq - We planned to do it by EOD tomorrow. I will update you here.

**Comment by laxmikantbpandhare** [comment]
@rbq - We have released Operator SDK v1.20.0. Could you please verify and let us know ur observations.

**Comment by laxmikantbpandhare** [comment]
@rbq - that's great, thank you for the confirmation.
